Cython/Compiler/ExprNodes.py

```diff
@@ -9488,15 +9488,22 @@ def analyse_default_args(self, env):
         if default_args or default_kwargs:
             if self.defaults_struct is None:
                 if default_args:
-                    defaults_tuple = TupleNode(self.pos, args=[
-                        arg.default for arg in default_args])
+                    # If the default argument has a children named arg, pass it down to the TupleNode.
```

Comment:
Haven't looked into the details, but this is a bit too hackish for a fix. We need to be more specific than "happens to have an attribute with name `arg`".

Reply:
Thanks to @da-woods this should now be much less hacky.